### PR TITLE
cherry picking other fixes out of the tracker scrollbar pr

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers 
had these fixes in https://github.com/keybase/client/pull/2428 but that's not moving anytime soon so let's split that out

### CODE REVIEW

File: desktop/app/store-helper.js [hunk 1]
```diff
@@ -20,7 +20,10 @@ export default function (mainWindow) {
     subscribeStoreSubscribers.push({sender, selector})
 
     try {
-      sender.send('stateChange', selector ? selector(store) : store)
+      const newStore = selector ? selector(store) : store
+      if (newStore) {
```
Comment (chrisnojima) [inline desktop/app/store-helper.js]: don't send updates if the store is just not there, likely due to a timing problem

File: shared/tracker/header.render.desktop.js [hunk 2]
```diff
@@ -31,7 +31,7 @@ export default class HeaderRender extends Component {
 
     let headerText: string = this.props.reason
     let isWarning = false
-    if (!this.props.currentlyFollowing && this.state.showCloseWarning) {
+    if (this.props.loggedIn && !this.props.currentlyFollowing && this.state.showCloseWarning) {
```
Comment (chrisnojima) [inline shared/tracker/header.render.desktop.js]: only show this warning if you're logged in

File: shared/tracker/index.js [hunk 3]
```diff
@@ -131,15 +132,19 @@ export default connect(
...
+        tracker: {
+          trackers: {
+            [username]: store.tracker.trackers[username]
+          }
+        },
+        config: store.config
+      }
     }
+
+    return null
```
Comment (chrisnojima) [inline shared/tracker/index.js]: don't send updates if we're not in the store for some reason yet

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by cjb** [comment]
:thumbsup:

**Pull request merged**